protocols/delegate-manager/test/DelegateManager-unit.js

```diff
@@ -0,0 +1,391 @@
+      .encodeABI()
+    await mockDelegate.givenMethodReturnBool(mockDelegate_unsetRule, true)
+
+    //mock owner()
+    let mockDelegate_owner = mockDelegateTemplate.contract.methods
+      .owner()
+      .encodeABI()
+    await mockDelegate.givenMethodReturnAddress(mockDelegate_owner, owner)
+  }
+
+  async function setupMockFactory() {
+    mockFactory = await MockContract.new()
+    let mockFactoryTemplate = await DelegateFactory.new(mockSwap.address)
+
+    // mock createDelegate()
+    let mockFactory_createDelegate = mockFactoryTemplate.contract.methods
+      .createDelegate(EMPTY_ADDRESS, EMPTY_ADDRESS)
+      .encodeABI()
+    await mockFactory.givenMethodReturnAddress(
+      mockFactory_createDelegate,
+      mockDelegate.address
+    )
+  }
+
+  async function setupMockIndexer() {
+    mockIndexer = await MockContract.new()
+    let mockIndexerTemplate = await Indexer.new(EMPTY_ADDRESS, EMPTY_ADDRESS)
+
+    //mock setIntent()
+    let mockIndexer_setIntent = mockIndexerTemplate.contract.methods
+      .setIntent(EMPTY_ADDRESS, EMPTY_ADDRESS, 0, web3.utils.fromAscii(''))
+      .encodeABI()
+    await mockIndexer.givenMethodReturnBool(mockIndexer_setIntent, true)
+
+    //mock unsetIntent()
+    let mockIndexer_unsetIntent = mockIndexerTemplate.contract.methods
+      .unsetIntent(EMPTY_ADDRESS, EMPTY_ADDRESS)
+      .encodeABI()
+    await mockIndexer.givenMethodReturnBool(mockIndexer_unsetIntent, true)
+
+    //mock stakeToken()
+    let mockIndexer_stakeToken = mockIndexerTemplate.contract.methods
+      .stakeToken()
+      .encodeABI()
+    await mockIndexer.givenMethodReturnAddress(
+      mockIndexer_stakeToken,
+      mockStakeToken.address
+    )
+
+    //mock getScore()
+    mockIndexer_getScore = mockIndexerTemplate.contract.methods
+      .getScore(EMPTY_ADDRESS, EMPTY_ADDRESS, EMPTY_ADDRESS)
+      .encodeABI()
+  }
+
+  before(async () => {
+    await setupMockTokens()
+    await setupMockSwap()
+    await setupMockIndexer()
+    await setupMockDelegate()
+    await setupMockFactory()
+
+    delegateManager = await DelegateManager.new(
+      mockFactory.address,
+      tradeWallet
+    )
+  })
+
+  describe('Test initial values', async () => {
+    it('Test factory address', async () => {
+      let val = await delegateManager.factory.call()
+      equal(val, mockFactory.address, 'mockFactory was not properly set')
+    })
+
+    it('Test delegate address', async () => {
+      let val = await delegateManager.delegate.call()
+      equal(val, mockDelegate.address, 'delegate address was not properly set')
+    })
+  })
+
+  describe('Test setRuleAndIntent()', async () => {
+    it('Test calling setRuleAndIntent with sender token mismatch', async() => {
+      let delegateAddress = await delegateManager.delegate.call()
+      let indexerAddress = mockIndexer.address
+
+      let rule = [mockWETH.address, mockWETH.address, 100000, 300, 0]
+
+      let intentAmount = 250
+      let intent = [
+        mockWETH.address,
```

Comment:
this is the wrong way round compared to the `let rule` above. Rule has senderToken then signerToken, whereas Intent is signerToken then senderToken.

Reply:
oh i see youve done that on purpose lol... shouldve read the whole test sorry